### PR TITLE
docs `branches.md`: explain the new defaults and reorganize 

### PR DESCRIPTION
**Pull request by ilyagr**
This follows up on #2625 and updates all sections of `branches.md` to represent the new default of `git.auto-local-branch=false`.

I also realized that the paragraph we discussed for so long with @PhilipMetzger actually repeats information that was already present a bit earlier in the file. So, I removed most of it and moved the rest. Sorry I didn't notice this earlier.

### CODE REVIEW

File: docs/branches.md [hunk 1]
```diff
@@ -11,23 +11,28 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+## Remotes and tracking branches
+
+Jujutsu records the last seen position of a branch on each remote (just like
+Git's remote-tracking branches). This record is updated on every `jj git fetch`
+and `jj git push` of the branch. You can refer to the remembered remote branch
+positions with `<branch name>@<remote name>`, such as `jj new main@origin`. `jj`
+does not provide a way to manually edit these records.
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked branch**. When you pull a tracking branch from a remote, any
```
Comment (ilyagr) [inline docs/branches.md:23]: TODO: Make sure this is in the glossary.

File: docs/branches.md [hunk 2]
```diff
@@ -11,23 +11,28 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+and `jj git push` of the branch. You can refer to the remembered remote branch
+positions with `<branch name>@<remote name>`, such as `jj new main@origin`. `jj`
+does not provide a way to manually edit these recorded positions.
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked branch**. When you pull a tracking branch from a remote, any
+changes compared to the current record of the remote's state will be propagated
+to the tracking local branch.
+
+**Details:** Let's say you run `jj git fetch --remote origin`, and and `jj`
```
Comment (ilyagr) [inline docs/branches.md:27]: I'd like to somehow mark this as "skip on first reading". `**Details**` is the best idea I've had for this so far.

File: docs/branches.md [hunk 3]
```diff
@@ -11,23 +11,28 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+## Remotes and tracked branches
+
+Jujutsu records the last seen position of a branch on each remote (just like
+Git's remote-tracking branches). This record is updated on every `jj git fetch`
+and `jj git push` of the branch. You can refer to the remembered remote branch
+positions with `<branch name>@<remote name>`, such as `jj new main@origin`. `jj`
+does not provide a way to manually edit these recorded positions.
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked branch**. When you pull a tracking branch from a remote, any
```
Comment (yuja) [inline docs/branches.md:23]: fwiw, I'm not sure which one we should call a tracking or tracked.

Appears that Git defines these as follows:
* remote-tracking branch: remote branches stored in local repo
* tracking branch: local branch tracking a remote branch
* upstream branch: remote-tracking branch tracked by a local branch
https://git-scm.com/book/en/v2/Git-Branching-Remote-Branches#_tracking_branches

Perhaps, we should avoid using "remote-tracking branch" because it's super confusing.

Comment (ilyagr) [reply docs/branches.md:23]: I chose "tracked" in this pr to decrease confusion with remote-tracking.

Maybe we need a different word entirely. "Synced?" "Followed remote branch and following local branch"?

Comment (ilyagr) [reply docs/branches.md:23]: For Git people: "downstreamed remote branch" or "mirrored downstream"

Comment (yuja) [reply docs/branches.md]: > I chose "tracked" in this pr to decrease confusion with remote-tracking.

I think "tracked" (for the remote counterpart) is pretty okay. The command is `jj branch track <branch>@<remote>`, so the remote branch can be considered "tracked".

"Synced" sounds a bit odd to me because the branch can get out of sync. Btw, I use "synced" (internally in templater) to represent a branch that doesn't have `*` sigil.

Comment (ilyagr) [reply docs/branches.md]: My one worry about "tracked" is that "local tracking branches" become inevitable and confusable with "remote-tracking branches". Perhaps it's ok if we always insert the word "local".

Comment (PhilipMetzger) [reply docs/branches.md]: Just commenting from the sidelines, I think "local tracking branches" is fine. Ideally we should have a term which doesn't overlap with Git.

File: docs/branches.md (hunk 2 above)
Comment (PhilipMetzger) [reply docs/branches.md:27]: Use a [alert](https://github.com/orgs/community/discussions/16925#discussion-4085374) here?

Comment (ilyagr) [reply docs/branches.md:27]: I'd love to, but unfortunately it seems that MkDocs uses a different syntax: https://squidfunk.github.io/mkdocs-material/reference/admonitions/. I should check whether they have an FR to support GitHub syntax.

Comment (ilyagr) [reply docs/branches.md:27]: On second thought, maybe I'll just use their syntax.

File: docs/branches.md (hunk 3 above)
Comment (ilyagr) [reply docs/branches.md]: I added a commit with an explanation of all of these terms. We can still change the terminology if we come up with something.

File: docs/branches.md (hunk 2 above)
Comment (ilyagr) [reply docs/branches.md:27]: The callout now looks like this on the website:

image

Comment (PhilipMetzger) [reply docs/branches.md]: See https://martinvonz.github.io/jj/v0.13.0/github/#adding-new-commits, to see how the alert in [github.md](https://github.com/martinvonz/jj/blob/main/docs/github.md) is rendered

File: docs/glossary.md [hunk 4]
```diff
@@ -173,6 +173,16 @@ A tree object represents a snapshot of a directory in the repository. Tree
 objects are defined recursively; each tree object only has the files and
 directories contained directly in the directory it represents.
 
+## Tracked branches and tracking branches
+A remote branch can be made "tracked" with the `jj track` command. This results
```
Comment (PhilipMetzger) [inline docs/glossary.md:177]: Newline above this for consistency with the rest of our docs.

File: docs/branches.md (hunk 2 above)
Comment (ilyagr) [reply docs/branches.md]: Interesting. I hadn't scrolled down enough on the link you provided, and thus haven't realized GitHub has two syntaxes: `> [!NOTE]` and just `> **Note**`. The latter does look OK in MkDocs, though it's annoying it can't have a period afterwards.

I have a few minor problems with the latter syntax:

- I'm not sure "Note" is the right type of a callout for this. GitHub documents it as "Highlights information that users should take into account, even when skimming.", and I was going for the opposite effect. I also don't see a better option among what GitHub provides, the closest is "Tip".
- GitHub considers the `> **Note**` syntax obsolete, though they don't seem in a hurry to remove it.

So, I think I still prefer what I have now? I'm not sure.

File: docs/branches.md [hunk 5]
```diff
@@ -52,6 +52,26 @@ some remote.
...
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
+  sync with the tracked remote branch. For example, after `jj branch track
+  develop@origin`, there will be a local branch `develop` that's tracking the
+  remote `develop@origin` branch.
+- A **remote-tracking branch** is a term common in the Git world, but `jj` has
+  them too. The term corresponds to the label of the last-remembered position of
+  a remote branch. For example, this is the commit that `jj checkout
```
Comment (PhilipMetzger) [inline docs/branches.md:68]: For me it was hard to parse the second sentence, as it is not clear about which VCS we're talking about. Can you make it a bit more explicit that the term description is for `jj`. 

While I was writing the above, I noticed this: isn't the `remote-tracking branch` just a bookmark to the last known remote commit? So wouldn't the term `remote-tracking branch-ref` be more accurate?

**Review by PhilipMetzger — CHANGES_REQUESTED** [review]
Minor nits, w/o bikeshedding about names.

File: docs/branches.md (hunk 2 above)
Comment (PhilipMetzger) [reply docs/branches.md]: I don't mind either syntax, but it would be nice to have it displayed uniformly in Markdown and on the website. In the end we'll need to chose one of them.

File: docs/branches.md (hunk 5 above)
Comment (ilyagr) [reply docs/branches.md:68]: > For me it was hard to parse the second sentence, as it is not clear about which VCS we're talking about. Can you make it a bit more explicit that the term description is for jj.

I can't think of an improvement right now, will look at it later. Perhaps I should be less aggressive about labeling this as a "Git term".

> While I was writing the above, I noticed this: isn't the remote-tracking branch just a bookmark to the last known remote commit? So wouldn't the term remote-tracking branch-ref be more accurate?

In my mind, the answer to your first question is "yes". I think of "branches" as a kind of "ref", and thus equivalent to "branch-refs". So, the term "remote-tracking branch-ref" sounds accurate to me, but completely equivalent, and I'm not sure it's clearer.

It's good you pointed this out; it does feel like a weird point of terminology that can confuse people with a different mindset from mine. Again, right now, I'm not sure how to improve it.

File: docs/branches.md [hunk 6]
```diff
@@ -11,50 +11,46 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+## Remotes and tracked branches
+
+Jujutsu records the last seen position of a branch on each remote (just like
+Git's remote-tracking branches). This record is updated on every `jj git fetch`
+and `jj git push` of the branch. You can refer to the remembered remote branch
+positions with `<branch name>@<remote name>`, such as `jj new main@origin`. `jj`
+does not provide a way to manually edit these recorded positions.
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked branch**. When you pull a tracked branch from a remote, any
```
Comment (yuja) [inline docs/branches.md:23]: nit: "called a tracked remote branch" for clarity?

File: docs/branches.md [hunk 7]
```diff
@@ -11,50 +11,46 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+changes compared to the current record of the remote's state will be propagated
+to the tracking local branch.
```
Comment (yuja) [inline docs/branches.md:25]: nit: maybe we can say the "corresponding local branch" or "associated ..." to avoid introducing the "tracking branch" terminology.

I think "tracking local branch" is confusing because the purpose of local branches isn't just for tracking.

File: docs/branches.md [hunk 8]
```diff
@@ -52,6 +52,26 @@ some remote.
...
+- A **remote branch** is a branch ref on the remote. `jj` can find out its state
+  only when it's actively communicating with the remote. 
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
+  sync with the tracked remote branch. For example, after `jj branch track
+  develop@origin`, there will be a local branch `develop` that's tracking the
+  remote `develop@origin` branch.
+- A **remote-tracking branch** is a term common in the Git world, but `jj` has
```
Comment (yuja) [inline docs/branches.md:66]: I would remove "remote-tracking branch" from the list because it's confusing. Perhaps, it can be merged into the "remote branch" definition.

We won't need to distinguish the remote branches in the local repo from the local branches in the remote repo in many contexts.

File: docs/branches.md (hunk 5 above)
Comment (PhilipMetzger) [reply docs/branches.md:68]: > I can't think of an improvement right now, will look at it later. Perhaps I should be less aggressive about labeling this as a "Git term".

Yes, that would be a improvement. Yuya has a good point in https://github.com/martinvonz/jj/pull/2772#discussion_r1442427271 too. 

> In my mind, the answer to your first question is "yes". I think of "branches" as a kind of "ref", and thus equivalent to "branch-refs". So, the term "remote-tracking branch-ref" sounds accurate to me, but completely equivalent, and I'm not sure it's clearer.

Currently branches are just "refs" for `jj`, as we don't have something which describes a named set of commits (like topics could be), and is the reason I used bookmark at first.  

I'd be in favor of calling it just the "last known remote revision" and then exposing it as such in the commit templater. 

> It's good you pointed this out; it does feel like a weird point of terminology that can confuse people with a different mindset from mine. Again, right now, I'm not sure how to improve it.

I don't think we have a huge missmatch in our concepts but as most users come from Git, for them the current phrasing is highly confusing. And since `jj` has nothing Git branch-like, this will only lead to more questions.

As @necauqua eloquently put it in Discord: 
> git gets away with it because it doesn't show non-bookmarked commits, so topological branch and the bookmark thing are kind of the same, which is slightly confusing on it's own btw when teaching people about stuff like hard resets

Git currently just get's away with.

File: docs/branches.md (hunk 7 above)
Comment (ilyagr) [reply docs/branches.md]: I tried to make it a bit better. I do feel that, if we have "tracked" branches, we should also explain what's tracking them.

File: docs/branches.md (hunk 8 above)
Comment (ilyagr) [reply docs/branches.md]: Done. I still discuss the distinction, but moving it into the same bullet point probably deepmhasizes it.

File: docs/branches.md (hunk 5 above)
Comment (ilyagr) [reply docs/branches.md]: I did deemphasize that it's a "Git term"

For the rest of this discussion, I think this is a little outside the scope of this PR. At least, I don't have an idea of how to improve it within the scope of the PR. There's an argument to be made that our "named branches" should not be called branches at all. I'm not sure that "ref" is a perfect name, since ideally I'd want to distinguish between refs that move with rebased commits and refs that don't, but there's an argument to be made that it's better than "branch".

However, in my mind, as long as we use the name "branches" following Git, mentioning "remote-tracking branches" is OK.

**Comment by ilyagr** [comment]
I remembered that this commit exists. I did some minor edits, I'm not sure they are sufficient. Let me know what you think or if you have better ideas.

File: docs/glossary.md [hunk 9]
```diff
@@ -173,6 +173,17 @@ A tree object represents a snapshot of a directory in the repository. Tree
 objects are defined recursively; each tree object only has the files and
 directories contained directly in the directory it represents.
 
+## Tracked branches and tracking branches
+
+A remote branch can be made "tracked" with the `jj track` command. This results
```
Comment (yuja) [inline docs/glossary.md:178]: `jj track` -> `jj branch track`

File: docs/branches.md [hunk 10]
```diff
@@ -11,50 +11,69 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+positions with `<branch name>@<remote name>`, such as `jj new main@origin`. `jj`
+does not provide a way to manually edit these recorded positions.
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked remote branch**. When you pull a tracked branch from a
+remote, any changes compared to the current record of the remote's state will be
+propagated to the corresponding local branch, which will be created if it
+doesn't exist already. (Currently, the tracking local branch always has the
+same name as the tracked remote branch; this restriction will likely be lifted
+in the future).
```
Comment (yuja) [inline docs/branches.md:28]: > (Currently, the tracking local branch always has the
same name as the tracked remote branch; this restriction will likely be lifted
in the future).

Just a comment. I personally don't have a plan to implement it because doing that would increase the UX complexity.

File: docs/branches.md [hunk 11]
```diff
@@ -11,50 +11,69 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+
+!!! note "Details: how `fetch` pulls branches"
+
+    Let's say you run `jj git fetch --remote origin`. Let's say that `jj` determines
+    that the remote's "main" branch has been moved so that its target is now ahead
```
Comment (yuja) [inline docs/branches.md:33]: > Let's say you run `jj git fetch --remote origin`. Let's say that `jj` determines

Maybe copy-paste error or missing something?

File: docs/branches.md [hunk 12]
```diff
@@ -11,50 +11,69 @@ pass a branch's name to commands that want a revision as argument. For example,
...
 
-Jujutsu's fetch operation consist of several steps. First `jj git fetch` fetches
-all Git refs under `refs/remotes/<remote name>`. Then Jujutsu stores these refs
-as remote tracking branches. Finally, if `git.auto-local-branch = true`, Jujutsu
-creates local branches for them. This is similar to Mercurial, which fetches all
-its bookmarks (equivalent to Git branches) by default.
+- A **remote branch** is a branch ref on the remote. `jj` can find out its
+  actual state only when it's actively communicating with the remote. However,
+  `jj` does store the last-seen position of the remote branch; this is the
+  commit `jj show develop@origin` would show. This notion is completely
```
Comment (yuja) [inline docs/branches.md:61]: nit: might be better to use `main@origin` as an example (because `develop@` doesn't appear in anywhere else.)

File: docs/branches.md [hunk 13]
```diff
@@ -11,50 +11,69 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+  analogous to Git's "remote-tracking branches".
+- A **tracked (remote) branch** is defined above. You can make a remote branch
+  tracked with the [`jj branch track` command](#manually-tracking-a-branch), for
+  example.
+- A **local tracking branch** is the local branch that `jj` tries to keep in
```
Comment (yuja) [inline docs/branches.md:66]: Perhaps, it should be "tracking local branch" or "tracking (local) branch". "local tracking" sounds like a branch that tracks local.

I still think it's better to avoid defining "tracking local branch" as a single term because "tracking" is just a state, but I don't have good suggestion here.

**Review by yuja — APPROVED** [review]
I'm not a good doc reviewer, but I don't find things obviously wrong or confusing. Thanks.

File: docs/branches.md (hunk 5 above)
Comment (PhilipMetzger) [reply docs/branches.md]: > I did deemphasize that it's a "Git term"

Thanks. 

> For the rest of this discussion, I think this is a little outside the scope of this PR.

Agreed. 

> There's an argument to be made that our "named branches" should not be called branches at all. I'm not sure that "ref" is a perfect name, since ideally I'd want to distinguish between refs that move with rebased commits and refs that don't, but there's an argument to be made that it's better than "branch".

I honestly think our "named branches" should just be called bookmarks, as martin proposed. The term "ref" mixes a technical (implementation) term with a user-visible term, which is honestly quite bad. All in all I think `jj` does not have something like Git's branches. 

> However, in my mind, as long as we use the name "branches" following Git, mentioning "remote-tracking branches" is OK.

I agree. I think we can put a lid on the "branch" naming conversation here and move it either to Discord or a issue.

**Review by PhilipMetzger — APPROVED** [review]
LG, since the latest edit.

File: docs/branches.md (hunk 10 above)
Comment (ilyagr) [reply docs/branches.md:28]: Good to know. I'll repeat this in the definitions then, if we don't think this is temporary.

File: docs/branches.md (hunk 13 above)
Comment (ilyagr) [reply docs/branches.md:66]: "tracking (local) branch" SGTM, I think.

File: docs/branches.md (hunk 12 above)
Comment (ilyagr) [reply docs/branches.md:61]: I think I didn't want to use `main@origin` because it's sometimes tracked by default. How about `mybranch`?

File: docs/branches.md [hunk 14]
```diff
@@ -76,9 +95,9 @@ $ jj new <name> # Do some local testing, etc.
 
 ### Untracking a branch
```
Comment (ilyagr) [inline docs/branches.md:97]: @yuja, @PhilipMetzger , added one more edit to the PR.

File: docs/branches.md (hunk 10 above)
Comment (ilyagr) [reply docs/branches.md:28]: I also noticed that this paragraph already says "of the same name", so I moved this to the "Terminology summary".

File: docs/branches.md (hunk 11 above)
Comment (ilyagr) [reply docs/branches.md:33]: No, that's just my writing. I added the word "also" for now, might reword it again later.

**Comment by ilyagr** [comment]
> I'm not a good doc reviewer, but I don't find things obviously wrong or confusing. Thanks.

Thank you! I found your review extremely helpful. Philip's review was also very helpful, thanks!

I'll leave this unmerged for a bit (a day?), in case somebody wants to proofread or look at the latest changes.

File: docs/branches.md (hunk 14 above)
Comment (ilyagr) [reply docs/branches.md:96]: I edited this some more and split these edits into a separate commit.

File: docs/branches.md (hunk 12 above)
Comment (yuja) [reply docs/branches.md:59]: Hmm, how about `jj show <branch name>@<remote name>`? I find it's a bit confusing because we're talking about "a remote branch" in general, but suddenly a specific example came in.

In the "tracking local branch" paragraph, `mybranch@origin` seems good since it's introduced as an example.

Comment (ilyagr) [reply docs/branches.md:59]: > Hmm, how about jj show <branch name>@<remote name>? I find it's a bit confusing because we're talking about "a remote branch" in general, but suddenly a specific example came in.

Sure, good idea.

File: docs/branches.md [hunk 15]
```diff
@@ -11,48 +11,45 @@ pass a branch's name to commands that want a revision as argument. For example,
...
+
+A remote branch can be associated with a local branch of the same name. This is
+called a **tracked remote branch**. When you pull a tracked branch from a
+remote, any changes compared to the current record of the remote's state will be
+propagated to the corresponding local branch, which will be created if it
+doesn't exist already.
+
+!!! note "Details: how `fetch` pulls branches"
+
+    Let's say you run `jj git fetch --remote origin`. Let's also say that `jj` determines
```
Comment (PhilipMetzger) [inline docs/branches.md:30]: Nit: Can you keep the first two sentences logically consistent? Something like "Let's say you run `jj git fetch --remote origin`. And after the fetch `jj` determines ..."

File: docs/glossary.md [hunk 16]
```diff
@@ -173,6 +173,17 @@ A tree object represents a snapshot of a directory in the repository. Tree
 objects are defined recursively; each tree object only has the files and
 directories contained directly in the directory it represents.
 
+## Tracked branches and tracking branches
+
+A remote branch can be made "tracked" with the `jj branch track` command. This
+results in a "tracking" local branch that tracks the remote branch.
+
+Both of these are different from Git's notion of "remote-tracking branches",
+though `jj` has an analogue to that as well.
```
Comment (PhilipMetzger) [inline docs/glossary.md:182]: Nit: Either I'm stupid and don't get the analogue from the existing text or this sentence lacks context. If you want to keep the sentence, refering to the analogue would be helpful.

**Review by PhilipMetzger — APPROVED** [review]
These are my final nits. Thanks for expanding my lazy examples and improving the rest so much.

File: docs/glossary.md (hunk 16 above)
Comment (ilyagr) [reply docs/glossary.md:182]: Good catch; I got rid of it.

File: docs/branches.md [hunk 17]
```diff
@@ -86,9 +87,17 @@ $ # Find the branch we no longer want to track.
...
 # # Actually untrack it.
 $ jj branch untrack <branch name>@<remote name> # Example: jj branch untrack stuff@origin
-$ # From this point on, it won't be imported anymore. 
+$ # From this point on, this remote branch won't be imported anymore.
+$ # The local branch (e.g. stuff) is unaffected. It may or may not still
+$ # be tracking branches on other remotes (e.g. stuff@upstream).
 ```
 
+After the `jj branch untrack`, you can forget the local branch with `jj branch
+forget`, if you wish. (Forgetting is safer than `jj branch delete`, since pushing
```
[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/branches.md:117]: I think I'll postpone this paragraph to a future PR (hopefully forthcoming). I would like to understand the behavior of `jj branch delete` + `jj git push` for untracked branches better.

File: docs/branches.md (hunk 15 above)
Comment (ilyagr) [reply docs/branches.md]: Yeah, Yuya was also unhappy with this phrasing. Here's another try.

**Comment by ilyagr** [comment]
Thanks for looking over my recent edits, Philip! Thank you also for starting the whole process of updating these docs.

File: docs/branches.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (yuja) [reply docs/branches.md]: Just fyi, pushing a deleted/existing-but-not-tracking branch to untracked remote is rejected. The underlying model is that push is allowed only if local and remote refs have the same merge base.

[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/branches.md]: Thanks, Yuya! This seems to work well, I'll think about how to document it.

Is it also true that `jj branch forget` automatically untracks a branch from all remotes? I was pleasantly surprised that `jj git push --branch qq; jj branch forget qq; jj git fetch` did not recreate the branch `qq`, but I wanted to double-check.

[the remark this answers is not in the document]
Comment (yuja) [reply docs/branches.md]: > Is it also true that `jj branch forget` automatically untracks a branch from all remotes?

Correct. `jj branch forget` discards all records about the branch, so re-fetched branch is completely new after that.

FWIW, it's undecided whether or not `jj branch forget branch` should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by `jj branch forget branch@remote`.

[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/branches.md]: > FWIW, it's undecided whether or not jj branch forget branch should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by jj branch forget branch@remote.

I think this makes sense.

Another thing `jj branch forget br` could do is to forget the local branch `br` and do the equivalent of `jj untrack glob:br@*`. This is close to what `jj branch forget; jj fetch` currently does. Perhaps this could be called `jj branch forget --local br`, or perhaps this should be the default behavior and the current one should be an option, I'm unsure.

[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/branches.md]: > FWIW, it's undecided whether or not jj branch forget branch should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by jj branch forget branch@remote.

On second thought, I think the user might want to forget them all at once. So, we should allow `jj branch forget glob:'br@*'`. I'm thinking this could work well with making the `--local` behavior I described the default. We could also have `jj branch forget --with-tracked-remotes br`.
